From 0930f0645f03f7bd59537bbb4550f62d5fe58a6a Mon Sep 17 00:00:00 2001
From: Alejandro Gonzalez <agonzalez@dextratech.com>
Date: Sat, 27 Nov 2010 16:07:19 -0600
Subject: [PATCH] qtmux: gst_qtmux_check_difference verify before subtract

Avoid negative overflow by checking the order of operands
on subtraction of unsigned integers.

https://bugzilla.gnome.org/show_bug.cgi?id=635878
---
 gst/qtmux/gstqtmux.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gst/qtmux/gstqtmux.c b/gst/qtmux/gstqtmux.c
index e138869..f2ad775 100644
--- a/gst/qtmux/gstqtmux.c
+++ b/gst/qtmux/gstqtmux.c
@@ -1904,7 +1904,7 @@ static gboolean inline
 gst_qtmux_check_difference (GstQTMux * qtmux, GstClockTime a,
     GstClockTime b, GstClockTime magn)
 {
-  return ((a - b >= (magn >> 1)) || (b - a >= (magn >> 1)));
+  return ((a >= b) ? (a - b >= (magn >> 1)) : (b - a >= (magn >> 1)));
 }
 
 /*
-- 
1.7.1

